### DISCUSSION TITLE
supports unnest for `map`

### DISCUSSION THREAD

**Discussion by StLeoX** [duckdb/duckdb#8452] [Ideas]
This requirement comes from #2876.
We expect this usage: `unnest(histogram(i))`.
The output of `histogram(i)` is `map` data type.

**Comment by Mause** [comment]
Does

```sql
unnest(map_entries(histogram()))
```
Return what you expect?

**Comment by StLeoX** [comment]
Doesn't work, there is an exception.

```
create table integers(i integer);
INSERT INTO integers VALUES (1),(1),(3),(20),(20),(20);
select unnest(map_entries(histogram(i))) FROM integers; 
```

Output:
Error: INTERNAL Error: Assertion triggered in file "/root/Source/cpp/duckdb/src/execution/operator/projection/physical_unnest.cpp" on line 136: child_vector.GetType() == result.GetType()

**Comment by StLeoX** [comment]
In this case, we expect output:
```
│ 1  │ 2        │
│ 3  │ 1        │
│ 20 │ 3        │
```
If using "unnest(map_entries(histogram(i)))", should add "recursive := true".

**Comment by StLeoX** [comment]
And generally, applying unnnest operation to map type should be allow.
